File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/RootServiceReceiver.java
package com.nikhilparanjape.radiocontrol.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.support.v4.content.WakefulBroadcastReceiver;
import android.util.Log;

import com.nikhilparanjape.radiocontrol.rootUtils.Utilities;
import com.nikhilparanjape.radiocontrol.services.CellRadioService;

/**
 * Created by admin on 9/24/2016.
 */

public class RootServiceReceiver extends WakefulBroadcastReceiver{
    public static final int REQUEST_CODE = 35718;

    public RootServiceReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        Intent i = new Intent(context, CellRadioService.class);
        Utilities util = new Utilities();

        context.stopService(i);
        Log.d("RadioControl","CellService Stopped");
        util.cancelRootAlarm(context);
        Log.d("RadioControl","RootClock cancelled");
    }
}

File: RadioControl/app/src/main/java/com/nikhilparanjape/radiocontrol/receivers/RootServiceReceiver.kt
@file:Suppress("DEPRECATION")

package com.nikhilparanjape.radiocontrol.receivers

import android.content.Context
import android.content.Intent
import android.util.Log
import androidx.legacy.content.WakefulBroadcastReceiver
import com.nikhilparanjape.radiocontrol.services.CellRadioService
import com.nikhilparanjape.radiocontrol.utilities.AlarmSchedulers

/**
 * Created by admin on 9/24/2016.
 */

class RootServiceReceiver : WakefulBroadcastReceiver() {
    private var alarmUtil = AlarmSchedulers()

    override fun onReceive(context: Context, intent: Intent) {
        val i = Intent(context, CellRadioService::class.java)

        context.stopService(i)
        Log.d("RadioControl-root", "CellService Stopped")
        alarmUtil.cancelRootAlarm(context)
        Log.d("RadioControl-root", "RootClock cancelled")
    }

    companion object {
        const val REQUEST_CODE = 35718
    }
}
